src/primitive/defs.rs

```diff
@@ -2104,6 +2104,27 @@ primitive!(
     /// ex: 1_2_3 4 5_6_7
     ///   : dump⊢
     (0(0)[1], Dump, Stack, "dump"),
+    /// Stringify any value
+    ///
+    /// ex: # Experimental!
+    ///   : repr π
+    /// Use [&p][repr] to produce a representation that can be pasted directly into the
+    /// interpreter.
+    /// ex: # Experimental!
+    ///   : &p repr ↯2_2_2 0
+    /// ex: # Experimental!
+    ///   : &p repr {"Uiua" @A [1 2 3] □4}
+    ///
+    /// Append commas to whitespace for a more traditional look.
+    /// ex: # Experimental!
+    ///   : ↯2_2_3⇡12
+    ///   : &p ⍜⊜□⎏(⊂@,)∊," \n" repr
+    ///
+    /// [map]s need to me unmapped before calling [repr].
```

Comment:
Can you check for the presence of the `map_len` metadata field, then generate a `map` call appropriately?

Reply:
Okay. Automatically unmapping is doable, but putting `map` calls into strings gets very weird because the precedence changes based on the context. I'll see what I can do

Reply:
> Can you check for the presence of the `map_len` metadata field, then generate a `map` call appropriately?

added code to clear out empty rows for maps, but I'm not sure how to go about printing a `map` call in a way that doesn't depend on line execution order.
here's two ways I could think of:
![изображение](https://github.com/uiua-lang/uiua/assets/72646905/97c30f99-d5c3-47f6-9943-b06eaf9a7cbd)

Reply:
[this](https://www.uiua.org/pad?src=0_8_0__IyBFeHBlcmltZW50YWwhCgojIE5vcm1hbCBleGVjdXRvbiBvcmRlcgp7IkNhcm9sIgogIkFsaWNlIgogIkJvYiJ9CltbNCA1XQogWzMgOF0KIFsxMiAyXV0KbWFwCgojIEFycmF5IGV4ZWN1dGlvbiBvcmRlcgriiqJbCiAgbWFwCiAgW1s0IDVdCiAgIFszIDhdCiAgIFsxMiAyXV0KICB7IkNhcm9sIgogICAiQWxpY2UiCiAgICJCb2IifV0K) is what the problem is, to be specific

Reply:
> or I could print it as a box array: ![изображение](https://private-user-images.githubusercontent.com/72646905/303924458-836b7843-b029-409a-869e-0ea8976cdbc5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc2ODQyMTMsIm5iZiI6MTcwNzY4MzkxMywicGF0aCI6Ii83MjY0NjkwNS8zMDM5MjQ0NTgtODM2Yjc4NDMtYjAyOS00MDlhLTg2OWUtMGVhODk3NmNkYmM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjExVDIwMzgzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3YjJiYzg2Zjg4NzIwYjg2ZjUzYjFiOWEzYTE5ZTJhN2VlNWI2YmI3NzNlZjkxNDgzZjdhMTQzMzEyYjVmYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cRYIK2s1xnqH6dTOp0Tq3gXHJbW08qhCMg-kA5N-1EY) this doesn't look weird and still works with `insert` that's how it works for now

I think this is fine.